buildSrc/src/main/groovy/com.flipkart.varadhi.java-library-conventions.gradle

```diff
@@ -4,4 +4,29 @@ plugins {
 
     // Apply the java-library plugin for API and implementation separation.
     id 'java-library'
+
+    // Apply the maven-publish to allow deploying to maven artifactories
+    id 'maven-publish'
+}
+
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = "${projectGroupId}"
+            artifactId = "${project.name}"
+            version = "${version}"
+            from components.java
+
+            pom {
+                name = "${project.name}"
```

Comment:
Do we need other details as well here ?

Reply:
this is to generate pom.xml so that the jar plays well when including in maven projects

Reply:
we can add more details if required, but currently I don't think anything else is required